base_layer/wallet/src/wallet.rs

```diff
@@ -156,6 +156,7 @@ where
         key_manager_backend: X,
         shutdown_signal: ShutdownSignal,
         master_seed: CipherSeed,
+        app_metadata: Option<AppMetadata>,
```

Comment:
Don't make this an option that's fed in from outside. 
Let the wallet ask its own version with `const::APP_VERSION_NUMBER`